central/delegatedregistryconfig/delegator/delegator.go

```diff
@@ -0,0 +1,148 @@
+	return &delegatorImpl{
+		deleRegConfigDS:   deleRegConfigDS,
+		connManager:       connManager,
+		scanWaiterManager: scanWaiterManager,
+	}
+}
+
+type delegatorImpl struct {
+	// deleRegConfigDS for pulling the current delegated registry config.
+	deleRegConfigDS datastore.DataStore
+
+	// connManager for sending scan requests to secured clusters and ensuring
+	// clusters are valid for delegation.
+	connManager connection.Manager
+
+	// scanWaiterManager creates waiters that wait for async scan responses.
+	scanWaiterManager waiter.Manager[*storage.Image]
+}
+
+// GetDelegateClusterID returns the cluster id that should enrich this image (if any) and
+// true if enrichment should be delegated to a secured cluster, false otherwise.
+func (d *delegatorImpl) GetDelegateClusterID(ctx context.Context, image *storage.Image) (string, bool, error) {
+	config, err := d.getConfig(ctx)
+	if err != nil {
+		return "", false, err
+	}
+
+	shouldDelegate, clusterID := d.shouldDelegate(image, config)
+	if !shouldDelegate {
+		return "", false, nil
+	}
+
+	err = d.validateCluster(clusterID)
+	return clusterID, true, err
+}
+
+// DelegateEnrichImage sends an enrichment request to the provided cluster.
+func (d *delegatorImpl) DelegateEnrichImage(ctx context.Context, image *storage.Image, clusterID string, force bool) error {
+	if clusterID == "" {
+		return errors.New("missing cluster id")
+	}
+
+	w, err := d.scanWaiterManager.NewWaiter()
+	if err != nil {
+		return err
+	}
+
+	msg := &central.MsgToSensor{
+		Msg: &central.MsgToSensor_ScanImage{
+			ScanImage: &central.ScanImage{
+				RequestId: w.ID(),
+				ImageName: image.GetName().GetFullName(),
+				Force:     force,
+			},
+		},
+	}
+
+	err = d.connManager.SendMessage(clusterID, msg)
+	if err != nil {
+		w.Close()
+		return err
+	}
+
+	log.Infof("Sent scan request %q to cluster %q for %q", w.ID(), clusterID, image.GetName().GetFullName())
+
+	img, err := w.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("error delegating scan to cluster %q for %q: %w", clusterID, image.GetName().GetFullName(), err)
+	}
+
+	log.Debugf("Scan response received for %q and image %q", w.ID(), img.GetName().GetFullName())
+
+	// Copy the fields from img into image, callers expecting image to be modified in place.
+	*image = *img
+
+	return nil
+}
+
+func (d *delegatorImpl) getConfig(ctx context.Context) (*storage.DelegatedRegistryConfig, error) {
+	config, _, err := d.deleRegConfigDS.GetConfig(ctx)
+	return config, err
+}
+
+func (d *delegatorImpl) shouldDelegate(image *storage.Image, config *storage.DelegatedRegistryConfig) (bool, string) {
+	if config == nil || config.GetEnabledFor() == storage.DelegatedRegistryConfig_NONE {
+		return false, ""
+	}
+
+	var should bool
+	if config.GetEnabledFor() == storage.DelegatedRegistryConfig_ALL {
+		should = true
+	}
+
+	clusterID := config.GetDefaultClusterId()
+	imageFullName := urlfmt.TrimHTTPPrefixes(image.GetName().GetFullName())
```

Comment:
Hrm, you may be right. I've debugged issues matching images to registries from pull secrets due to some pull secrets having prefixes and others not. 

It doesn't appear an imageName (if using the utility methods in `pkg/image/utils`) can have a prefix however. If we're OK assuming an ImageName is always populated using the utils and will never have a prefix, I'm OK changing it - otherwise would consider this defensive code.

Reply:
Did some more research and can confirm that the execution path for `roxctl deployment check` does pass the image string as it exists in the `yaml` to the image enricher.  The enrichments fails later on due to parsing errors. Based on my preference would be to keep the trim.